src/lsptoolshost/dotnetRuntimeExtensionResolver.ts

```diff
@@ -101,4 +123,103 @@ export class DotnetRuntimeExtensionResolver implements IHostExecutableResolver {
 
         return dotnetInfo;
     }
+
+    /**
+     * Checks dotnet --version to see if the value on the path is greater than the minimum required version.
+     * This is adapated from similar O# server logic and should be removed when we have a stable acquisition extension.
+     * @returns true if the dotnet version is greater than the minimum required version, false otherwise.
+     */
+    private async findDotnetFromPath(): Promise<string | undefined> {
+        try {
+            const dotnetInfo = await getDotnetInfo([]);
+            const dotnetVersionStr = dotnetInfo.Version;
+
+            const extensionArchitecture = await this.getArchitectureFromTargetPlatform();
+            const dotnetArchitecture = dotnetInfo.Architecture;
+
+            // If the extension arhcitecture is defined, we check that it matches the dotnet architecture.
+            // If its undefined we likely have a platform neutral server and assume it can run on any architecture.
+            if (extensionArchitecture && extensionArchitecture !== dotnetArchitecture) {
+                throw new Error(
+                    `The architecture of the .NET runtime (${dotnetArchitecture}) does not match the architecture of the extension (${extensionArchitecture}).`
+                );
+            }
+
+            const dotnetVersion = semver.parse(dotnetVersionStr);
+            if (!dotnetVersion) {
+                throw new Error(`Unknown result output from 'dotnet --version'. Received ${dotnetVersionStr}`);
+            }
+
+            if (semver.lt(dotnetVersion, this.minimumDotnetVersion)) {
+                throw new Error(
```

Comment:
Would it make sense to issue a warning and fall back to the dotnet acquisition extension in this case?

Reply:
Nevermind. I see.